### PR TITLE
Add API to fetch replication message

### PR DESCRIPTION
**Pull request by yux0**
1. Add new API to get replication task using DLQ info
2. Fix mockgen 
3. Fix lint

### CODE REVIEW

File: idl/github.com/uber/cadence/admin.thrift [hunk 1]
```diff
@@ -105,6 +105,14 @@ service AdminService {
       5: shared.ClientVersionNotSupportedError clientVersionNotSupportedError,
     )
 
+  replicator.GetDLQReplicationMessagesResponse GetDLQReplicationMessages(1: replicator.GetDLQReplicationMessagesRequest request)
+    throws (
+      1: shared.BadRequestError badRequestError,
+      2: shared.LimitExceededError limitExceededError,
+      3: shared.ServiceBusyError serviceBusyError,
+      4: shared.ClientVersionNotSupportedError clientVersionNotSupportedError,
```
Comment (wxing1292) [inline idl/github.com/uber/cadence/admin.thrift]: this error is should not be here?

File: idl/github.com/uber/cadence/admin.thrift [hunk 2]
```diff
@@ -105,6 +105,14 @@ service AdminService {
       5: shared.ClientVersionNotSupportedError clientVersionNotSupportedError,
     )
 
+  replicator.GetDLQReplicationMessagesResponse GetDLQReplicationMessages(1: replicator.GetDLQReplicationMessagesRequest request)
+    throws (
+      1: shared.BadRequestError badRequestError,
+      2: shared.LimitExceededError limitExceededError,
```
Comment (wxing1292) [inline idl/github.com/uber/cadence/admin.thrift]: this error is should not be here?

File: idl/github.com/uber/cadence/history.thrift [hunk 3]
```diff
@@ -815,6 +819,19 @@ service HistoryService {
...
+  * GetDLQReplicationMessages return replication messages based on dlq info
+  **/
+  replicator.GetDLQReplicationMessagesResponse GetDLQReplicationMessages(1: replicator.GetDLQReplicationMessagesRequest request)
+    throws (
+      1: shared.BadRequestError badRequestError,
+      2: shared.InternalServiceError internalServiceError,
+      3: shared.LimitExceededError limitExceededError,
+      4: shared.ServiceBusyError serviceBusyError,
+      5: shared.ClientVersionNotSupportedError clientVersionNotSupportedError,
+      6: shared.EntityNotExistsError entityNotExistError,
```
Comment (wxing1292) [inline idl/github.com/uber/cadence/history.thrift]: same as above
when will EntityNotExistsError be returned

File: service/history/conflictResolver.go [hunk 4]
```diff
@@ -18,7 +18,7 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-//go:generate mockgen -copyright_file ../../LICENSE -package $GOPACKAGE -source $GOFILE -destination conflictResolver_mock.go
+//go:generate mockgen -copyright_file ../../LICENSE -package $GOPACKAGE -source $GOFILE -destination conflictResolver_mock.go -self_package github.com/uber/cadence/common/history -self_package github.com/uber/cadence/service/history
```
Comment (wxing1292) [inline service/history/conflictResolver.go]: duplicate `-self`

File: service/history/replicationTaskProcessor_test.go [hunk 5]
```diff
@@ -18,6 +18,8 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
+// +build test
```
Comment (wxing1292) [inline service/history/replicationTaskProcessor_test.go]: remove this

File: service/history/handler.go [hunk 6]
```diff
@@ -1479,6 +1480,79 @@ func (h *Handler) GetReplicationMessages(
...
+		taskInfoPerExecution[key] = append(taskInfoPerExecution[key], taskInfo)
+	}
+
+	var wg sync.WaitGroup
+	wg.Add(len(taskInfoPerExecution))
+	tasksChan := make(chan *r.ReplicationTask, len(request.GetTaskInfos()))
+	handleTaskInfoPerExecution := func(taskInfos []*r.ReplicationTaskInfo) {
+		defer wg.Done()
+
+		if len(taskInfos) > 0 {
```
Comment (wxing1292) [inline service/history/handler.go]:
```go
len(taskInfos) == 0 {
  return
}
```

File: idl/github.com/uber/cadence/admin.thrift (hunk 2 above)
Comment (yux0) [reply idl/github.com/uber/cadence/admin.thrift]: removed

File: idl/github.com/uber/cadence/admin.thrift (hunk 1 above)
Comment (yux0) [reply idl/github.com/uber/cadence/admin.thrift]: removed

File: idl/github.com/uber/cadence/history.thrift (hunk 3 above)
Comment (yux0) [reply idl/github.com/uber/cadence/history.thrift]: When the workflow passed the retention period and mutable state is not found

File: service/history/replicationTaskProcessor_test.go (hunk 5 above)
Comment (yux0) [reply service/history/replicationTaskProcessor_test.go]: removed

File: service/history/conflictResolver.go (hunk 4 above)
Comment (yux0) [reply service/history/conflictResolver.go]: removed

File: service/history/handler.go (hunk 6 above)
Comment (yux0) [reply service/history/handler.go]: updated

**Pull request merged**